pkg/kubelet/dockershim/network/kubenet/kubenet_linux.go

```diff
@@ -264,15 +264,15 @@ func (plugin *kubenetNetworkPlugin) Event(name string, details map[string]interf
 	}
 
 	for idx, currentPodCIDR := range podCIDRs {
-		_, cidr, err := net.ParseCIDR(currentPodCIDR)
+		ip, cidr, err := net.ParseCIDR(currentPodCIDR)
 		if nil != err {
 			klog.Warningf("Failed to generate CNI network config with cidr %s at indx:%v: %v", currentPodCIDR, idx, err)
 			return
 		}
 		// create list of ips and gateways
-		cidr.IP[len(cidr.IP)-1] += 1 // Set bridge address to first address in IPNet
 		plugin.podCIDRs = append(plugin.podCIDRs, cidr)
-		plugin.podGateways = append(plugin.podGateways, cidr.IP)
+		ip[len(ip)-1]++
```

Comment:
I thinks this operation is dangerous at some time.
Actually `host-local` has the ability to conclude an appropriate gateway for a subnet, just as https://github.com/containernetworking/plugins/blob/497560f35f2cef2695f1690137b0bba98adf849b/plugins/ipam/host-local/backend/allocator/range.go#L51
Why not just keep `subnet` and abandon `gateway` in network config?

/cc @dcbw

Reply:
> I thinks this operation is dangerous at some time.

If you see the first [commit](https://github.com/kubernetes/kubernetes/pull/84550/commits/a6ca51be6ebeae2c3e1b3a1d9a17a9b9f89d40a5#diff-fc2cc81fd1528b716e7ff3cebcd70db8R274) I was using "k8s.io/utils/net" to generate it but changed that based on the reviews

> Why not just keep subnet and abandon gateway in network config?

It's used later in the code to generate iptables rules 🤷‍♂ 

https://github.com/kubernetes/kubernetes/blob/cdc07c821a0d5d3a4d396a3df3e87e2bc5c10e81/pkg/kubelet/dockershim/network/kubenet/kubenet_linux.go#L763

Reply:
Oh, I'll buy in generating a gateway from a CIDR for ebtables, but I don't think `ip[len(ip)-1]++` will help us, see https://play.golang.org/p/Oa1thRsqIfS, that's an unexpected result.

Reply:
that was exactly my point in the first commit and what I've explained here
https://github.com/kubernetes/kubernetes/pull/84550#issuecomment-549570475
the argument I've received in the review is that the API will guarantee that the podCIDR will always be using the pod subnet address.
Should we use the approach in the first commit a6ca51be6ebeae2c3e1b3a1d9a17a9b9f89d40a5 ?

Reply:
Yes, IMO, the first commit is better.